presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -86,8 +86,8 @@ else if (javaType == Slice.class) {
         }
 
         try {
-            connection = jdbcClient.getConnection(split);
-            statement = jdbcClient.buildSql(connection, split, columnHandles);
+            connection = jdbcClient.getConnection(JdbcIdentity.from(session), split);
+            statement = jdbcClient.buildSql(session, connection, split, columnHandles);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *statement*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).